source/kernel/src/forth/mod.rs

```diff
@@ -35,6 +35,48 @@ pub struct Params {
     pub spawnulator_timeout: Duration,
 }
 
+#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
+#[non_exhaustive]
+pub struct ParamsOverrides {
+    pub stack_size: Option<usize>,
+    pub input_buf_size: Option<usize>,
+    pub output_buf_size: Option<usize>,
+    pub dictionary_size: Option<usize>,
+    pub stdin_capacity: Option<usize>,
+    pub stdout_capacity: Option<usize>,
+    pub bag_of_holding_capacity: Option<usize>,
+    pub spawnulator_timeout: Option<Duration>,
+}
+
+impl ParamsOverrides {
+    pub fn into_settings(self) -> Params {
+        Params {
+            stack_size: self.stack_size.unwrap_or(Params::DEFAULT_STACK_SIZE),
+            input_buf_size: self
+                .input_buf_size
+                .unwrap_or(Params::DEFAULT_INPUT_BUF_SIZE),
+            output_buf_size: self
+                .output_buf_size
+                .unwrap_or(Params::DEFAULT_OUTPUT_BUF_SIZE),
+            dictionary_size: self
+                .dictionary_size
+                .unwrap_or(Params::DEFAULT_DICTIONARY_SIZE),
+            stdin_capacity: self
+                .stdin_capacity
+                .unwrap_or(Params::DEFAULT_STDIN_CAPACITY),
+            stdout_capacity: self
+                .stdout_capacity
+                .unwrap_or(Params::DEFAULT_STDOUT_CAPACITY),
+            bag_of_holding_capacity: self
+                .bag_of_holding_capacity
+                .unwrap_or(Params::DEFAULT_BAG_OF_HOLDING_CAPACITY),
+            spawnulator_timeout: self
+                .spawnulator_timeout
+                .unwrap_or(Params::DEFAULT_SPAWNULATOR_TIMEOUT),
+        }
+    }
```

Comment:
i think there might be an easier way of handling the default values: https://serde.rs/attr-default.html